### ISSUE TITLE
Annotation classes don’t conform to NSCoding

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#6200]
The following classes don’t conform to NSSecureCoding but probably should (and could without much fuss):
- MGLAnnotationImage
- MGLAnnotationView
- MGLShape
- MGLUserLocation
- MGLUserLocationAnnotationView

The developer may want to archive an instance of one of these classes for state restoration or some other kind of persistence, but they’ll run into a missing implementation in doing so:

```
-[MGLUserLocation encodeWithCoder:]: unrecognized selector sent to instance 0x60800024b820
```

/cc @friedbunny

**Comment by 1ec5** [comment]
We might be able to generalize #6524 to cover all the shape subclasses, making this task a lot easier.

MGLUserLocationAnnotation is tricker to code because of its use of a delegate pattern and a weak backpointer to MGLMapView. We should just omit those problematic properties from the coded representation.

**Comment by 1ec5** [comment]
Work towards this issue is happening in #6559.

**Comment by 1ec5** [comment]
Fixed by #6559 on release-ios-v3.4.0 branch.
